mv/MBS_MobileDirPad.js

```diff
@@ -287,85 +725,118 @@ MBS.MobileDirPad = {};
 	Scene_Base.prototype.update = function() {
-		Scene_Base_update.apply(this, arguments);
-		if (this.isMobileDevice() && this._dirPad != undefined && this._aButton != undefined && this._cButton != undefined)
-			this._dirPad.visible = this._aButton.visible = this._cButton.visible = Scene_Base.dirpad;
+		MBS.MobileDirPad.Scene_Base_update.apply(this, arguments);
+		if (this.isMobileDevice() && this._dirPad != undefined && this._aButton != undefined && this._cButton != undefined && this._shiftButton != undefined && this._mButton != undefined && this._qButton != undefined && this._wButton != undefined)
+			this._dirPad.visible = this._aButton.visible = this._cButton.visible = this._shiftButton.visible = this._mButton.visible = this._qButton.visible = this._wButton.visible = Scene_Base.dirpad;
 	};
 
 	Scene_Base.prototype.createDirPad = function() {
 		this._dirPad = new Sprite_DirPad();
-		this._dirPad.opacity = $.Param.opacity;
+		this._dirPad.opacity = MBS.MobileDirPad.Param.opacity;
 
-		this._dirPad.x = $.Param.dpadPosition.x;
-		this._dirPad.y = $.Param.dpadPosition.y;
+		this._dirPad.x = MBS.MobileDirPad.Param.dPadX;
+		this._dirPad.y = MBS.MobileDirPad.Param.dPadY;
 
 		this.addChild(this._dirPad);
 	};
 
 	Scene_Base.prototype.createActionButtons = function() {
-		this._aButton = new Sprite_Button(0);
-		this._aButton.opacity = $.Param.opacity;
-
-		this._aButton.x = $.Param.buttonPosition.x;
-		this._aButton.y = $.Param.buttonPosition.y;
-
-		this._cButton = new Sprite_Button(1);
-		this._cButton.opacity = $.Param.opacity;
-
-		this._cButton.x = $.Param.cButtonPosition.x;
-		this._cButton.y = $.Param.cButtonPosition.y;
+		this._aButton = new Sprite_mbsButton(0);
+		this._aButton.opacity = MBS.MobileDirPad.Param.opacity;
+
+		this._aButton.x = MBS.MobileDirPad.Param.actionButtonX;
+		this._aButton.y = MBS.MobileDirPad.Param.actionButtonY;
+
+		this._cButton = new Sprite_mbsButton(1);
+		this._cButton.opacity = MBS.MobileDirPad.Param.opacity;
+
+		this._cButton.x = MBS.MobileDirPad.Param.cancelButtonX;
+		this._cButton.y = MBS.MobileDirPad.Param.cancelButtonY;
+		
+		this._shiftButton = new Sprite_mbsButton(2);
+		this._shiftButton.opacity = MBS.MobileDirPad.Param.opacity;
+
+		this._shiftButton.x = MBS.MobileDirPad.Param.shiftButtonX;
+		this._shiftButton.y = MBS.MobileDirPad.Param.shiftButtonY;
+		
+		this._mButton = new Sprite_mbsButton(3);
+		this._mButton.opacity = MBS.MobileDirPad.Param.opacity;
+
+		this._mButton.x = MBS.MobileDirPad.Param.menuButtonX;
+		this._mButton.y = MBS.MobileDirPad.Param.menuButtonY;
+		
+		this._qButton = new Sprite_mbsButton(4);
+		this._qButton.opacity = MBS.MobileDirPad.Param.opacity;
+
+		this._qButton.x = MBS.MobileDirPad.Param.qButtonX;
+		this._qButton.y = MBS.MobileDirPad.Param.qButtonY;
+		
+		this._wButton = new Sprite_mbsButton(5);
+		this._wButton.opacity = MBS.MobileDirPad.Param.opacity;
+
+		this._wButton.x = MBS.MobileDirPad.Param.wButtonX;
+		this._wButton.y = MBS.MobileDirPad.Param.wButtonY;
```

Comment:
Same thing as above for all these initializations. If the buttons were in an array/object it would be possible to just iterate over it and initialize them all at once.